gulpfile.js

```diff
@@ -85,6 +86,84 @@ function createReloadServer() {
   });
 }
 
+function writeServiceWorkerFile() {
+  let path = require('path');
+  let rootDir = 'dist';
+
+  let dynamicUrlToDependencies = {};
+  let partialTemplateFiles = ['head-meta.html', 'site-nav.html']
+    .map(file => path.join(rootDir, 'templates', file));
+
+  ['about', 'index', 'shell'].forEach(htmlFile => {
+    let dynamicUrl = '/' + htmlFile;
+    let dependencies = partialTemplateFiles.concat(`${rootDir}/${htmlFile}.html`);
+    dynamicUrlToDependencies[dynamicUrl] = dependencies;
+    // Treat / like /index
+    if (htmlFile === 'index') {
+      dynamicUrlToDependencies['/'] = dependencies;
+    }
+  });
+
+  let config = {
+    cacheId: 'polymerproject',
+    staticFileGlobs: [
+      `${rootDir}/images/logos/p-logo.png`,
+      `${rootDir}/elements/**`,
+      `${rootDir}/js/*.js`,
+      `${rootDir}/css/*.css`,
+      `${rootDir}/bower_components/**/webcomponents-lite.min.js`,
+      `${rootDir}/bower_components/highlight/highlight.js`,
+    ],
+    dynamicUrlToDependencies: dynamicUrlToDependencies,
+    navigateFallback: '/shell',
```

Comment:
It is possible to disable `navigateFallback` and then write your own custom logic (pulled in via `importScripts`) that responds to navigation requests using whatever logic makes the most sense.

It really sounds like you don't want to use the App Shell model at all, though...